lib/sen_tweet/bitfeels_bot/twitter.ex

```diff
@@ -10,39 +10,64 @@ defmodule SenTweet.BitfeelsBot.Twitter do
 
   delete_tweet - POST statuses/destroy/:id
   https://developer.twitter.com/en/docs/tweets/post-and-engage/api-reference/post-statuses-destroy-id
+
+  upload_image - POST media/upload (Image Size 5MB MAX)
+  https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/api-reference/post-media-upload
   """
 
   alias SenTweet.BitfeelsBot.Auth
 
-  def tweet do
+  def tweet(tweet_text) do
+    method = "post"
+    url = "https://api.twitter.com/1.1/statuses/update.json"
+    params = [{"status", tweet_text}]
+    {auth_header, req_params} = Auth.auth_header(method, url, params)
+
+    do_request(url, [auth_header], req_params)
+  end
+
+  def tweet_with_image(tweet_text, media_id) do
     method = "post"
     url = "https://api.twitter.com/1.1/statuses/update.json"
-    params = [{"status", "hello twitter 🤖"}]
-    {header, req_params} = Auth.auth_header(method, url, params)
+    params = [{"status", tweet_text}, {"media_ids", "#{media_id}"}]
+    {auth_header, req_params} = Auth.auth_header(method, url, params)
 
-    do_request(url, header, req_params)
+    do_request(url, [auth_header], req_params)
   end
 
   def get_tweet(tweet_id) do
     method = "post"
     url = "https://api.twitter.com/1.1/statuses/lookup.json"
     params = [{"id", tweet_id}]
-    {header, req_params} = Auth.auth_header(method, url, params)
+    {auth_header, req_params} = Auth.auth_header(method, url, params)
 
-    do_request(url, header, req_params)
+    do_request(url, [auth_header], req_params)
   end
 
   def delete_tweet(tweet_id) do
     method = "post"
     url = "https://api.twitter.com/1.1/statuses/destroy/#{tweet_id}.json"
-    params = []
-    {header, req_params} = Auth.auth_header(method, url, params)
+    {auth_header, req_params} = Auth.auth_header(method, url, [])
+
+    do_request(url, [auth_header], req_params)
+  end
+
+  def upload_image(image_name) do
+    method = "post"
+    url = "https://upload.twitter.com/1.1/media/upload.json"
+
+    image_path = "images/" <> image_name
+    encoded_image = image_path |> File.read!() |> Base.encode64()
```

Comment:
Looks like this will make it feasible for us to generate graphs as image files, upload them, and then tweet with them. Will it be difficult to first upload an image and then post with it?

Reply:
We should be able to upload any image that's a png / jpg and <= 5MB.

No, we can programmatically upload then post.

There may be some API limitations on how much we can upload with the free version, I'll have to look into that. These types of services typically don't have a _hey you already uploaded this, here's the media ID_, they just upload a new one.